test/modules/aws-vpc-routetable-integration.ts

```diff
@@ -0,0 +1,262 @@
+import { v4 as uuidv4 } from 'uuid';
+
+import * as Modules from '../../src/modules';
+import { awsAccount, awsVpcModule } from '../../src/modules';
+import { RouteTableAssociation } from '../../src/modules/aws_vpc/entity';
+import { getContext } from '../../src/router/db';
+import * as iasql from '../../src/services/iasql';
+import { TypeormWrapper } from '../../src/services/typeorm';
+import {
+  defaultRegion,
+  execComposeDown,
+  execComposeUp,
+  finish,
+  getPrefix,
+  runBegin,
+  runCommit,
+  runInstall,
+  runQuery,
+} from '../helpers';
+
+const dbAlias = 'iasql';
+const randIPBlock = Math.floor(Math.random() * 254) + 1; // 0 collides with the default CIDR block
+jest.setTimeout(360000);
+beforeAll(async () => await execComposeUp());
+afterAll(async () => await execComposeDown());
+
+const begin = runBegin.bind(null, dbAlias);
+const query = runQuery.bind(null, dbAlias);
+const install = runInstall.bind(null, dbAlias);
+const commit = runCommit.bind(null, dbAlias);
+const region = defaultRegion();
+
+const prefix = getPrefix();
+let username: string, password: string;
+
+describe('RouteTable Integration Testing', () => {
+  let context: { [x: string]: any };
+
+  it('creates a new test db', done => {
+    (async () => {
+      try {
+        const { user, password: pgPassword } = await iasql.connect(dbAlias, 'not-needed', 'not-needed');
+        username = user;
+        password = pgPassword;
+        if (!username || !password) throw new Error('Did not fetch pg credentials');
+        done();
+      } catch (e) {
+        done(e);
+      }
+    })();
+  });
+
+  it('inits variables', async () => {
+    const conn = await TypeormWrapper.createConn(dbAlias, { name: uuidv4() });
+    const initialContext = await getContext(conn, Modules);
+    context = { ...awsAccount.context, ...initialContext };
+  });
+
+  it('installs the aws_account module', install(['aws_account']));
+
+  it(
+    'inserts aws credentials',
+    query(
+      `
+          INSERT INTO aws_credentials (access_key_id, secret_access_key)
+          VALUES ('${process.env.AWS_ACCESS_KEY_ID}', '${process.env.AWS_SECRET_ACCESS_KEY}')
+      `,
+      undefined,
+      false,
+      () => ({ username, password }),
+    ),
+  );
+
+  it('starts a transaction', begin());
```

Comment:
why begin/commit here without any other queries?

Reply:
That's for syncing the region. We have it in other tests as well. I think we need to do it this way.

Reply:
ok i see, that looked weird to me

Reply:
Yes I've checked that. We can't do a `iasql_commit` without a `iasql_begin`.

Reply:
In the RFC for two-way mode we planned out an `iasql_wait_for` helper function that polls a record waiting for the specified column to be non-null. We haven't implemented that yet, partly because that syntax still feels a bit weird, but perhaps we should?